internal/pkg/service/common/configpatch/dump.go

```diff
@@ -41,80 +39,96 @@ func DumpKVs(configStruct, patchStruct any, opts ...Option) ([]DumpKV, error) {
 	// Visit patch, get patched values
 	patchKeys := make(map[string]bool)
 	patchedValues := make(map[string]reflect.Value)
-	visitStruct(cfg.nameTags, patchStruct, func(vc *configmap.VisitContext) {
-		// Process only leaf values with a field name
-		if !vc.Leaf || vc.MappedPath.Last().String() == "" {
-			return
-		}
+	configmap.MustVisit(
+		reflect.ValueOf(patchStruct),
+		configmap.VisitConfig{
+			OnField: matchTaggedFields(cfg.nameTags),
+			OnValue: func(vc *configmap.VisitContext) error {
```

Comment:
In same places we need `AfterValue` callback, but not in all.
Instead of a third nullable argument in the `visit struct`, I used `config struct ` approach.